src/streaming/array.writer.cpp

```diff
@@ -239,12 +239,12 @@ zarr::ArrayWriter::make_buffers_() noexcept
 
     for (auto& buf : chunk_buffers_) {
         buf.resize(nbytes);
-        std::fill_n(buf.begin(), nbytes, std::byte(0));
+        std::fill(buf.begin(), buf.end(), std::byte(0));
     }
 }
 
 size_t
-zarr::ArrayWriter::write_frame_to_chunks_(std::span<std::byte> data)
+zarr::ArrayWriter::write_frame_to_chunks_(std::span<const std::byte> data)
```

Comment:
Interesting choice to mark the underlying data of the span as const. What's the reasoning behind this?